src/webgl/ShaderGenerator.js

```diff
@@ -1638,3 +1638,141 @@ export default shadergenerator;
+ *
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * let myShader;
+ * function setup() {
+ *   createCanvas(200, 200, WEBGL);
+ *   myShader = baseMaterialShader().modify(() => {
+ *     getWorldInputs(inputs => {
+ *       // Move the vertex up and down in a wave
+ *       inputs.position.y += 20 * sin(
+ *         millis() * 0.001 + inputs.position.x * 0.05
+ *       );
+ *       return inputs;
+ *     });
+ *   });
+ * }
+ * function draw() {
+ *   background(255);
+ *   shader(myShader);
+ *   lights();
+ *   noStroke();
+ *   fill('red');
+ *   sphere(50);
+ * }
+ * </code>
+ * </div>
+ */
+
+/**
+ * @function combineColors
+ * @experimental
+ * @description
+ * Registers a callback to customize how color components are combined in the fragment shader. This hook can be used inside <a href="#/p5/baseMaterialShader">baseMaterialShader()</a>.modify() and similar shader modify calls to control the final color output of a material. The callback receives color components (baseColor, diffuse, ambientColor, ambient, specularColor, specular, emissive, opacity) and returns a vec4 for the final color.
+ *
+ * This hook is available in:
+ * - <a href="#/p5/baseMaterialShader">baseMaterialShader()</a>
+ * - <a href="#/p5/baseNormalShader">baseNormalShader()</a>
+ * - <a href="#/p5/baseColorShader">baseColorShader()</a>
+ * - <a href="#/p5/baseStrokeShader">baseStrokeShader()</a>
+ *
+ * @param {function} callback
+ *        A callback function which receives color components (baseColor, diffuse, ambientColor, ambient, specularColor, specular, emissive, opacity) and returns a vec4 for the final color.
+ *
+ * @example
+ * <div modernizr='webgl'>
+ * <code>
+ * let myShader;
+ * function setup() {
+ *   createCanvas(200, 200, WEBGL);
+ *   myShader = baseMaterialShader().modify(() => {
+ *     combineColors(components => {
+ *       // Custom color combination: add a red tint
+ *       let r = components.baseColor.r * components.diffuse.r +
+ *               components.ambientColor.r * components.ambient.r +
+ *               components.specularColor.r * components.specular.r +
+ *               components.emissive.r + 0.2;
+ *       let g = components.baseColor.g * components.diffuse.g +
+ *               components.ambientColor.g * components.ambient.g +
+ *               components.specularColor.g * components.specular.g +
+ *               components.emissive.g;
+ *       let b = components.baseColor.b * components.diffuse.b +
+ *               components.ambientColor.b * components.ambient.b +
+ *               components.specularColor.b * components.specular.b +
+ *               components.emissive.b;
+ *       let a = components.opacity;
+ *       return vec4(r, g, b, a);
+ *     });
+ *   });
+ * }
+ * function draw() {
+ *   background(255);
+ *   shader(myShader);
+ *   lights();
+ *   noStroke();
+ *   fill('red');
+ *   sphere(50);
+ * }
+ * </code>
+ * </div>
+ */
+
+/**
+ * @function getPointSize
```

Comment:
Just a heads up -- currently the only public base shaders are:

- [baseColorShader()](https://beta.p5js.org/reference/p5/baseColorShader)
- [baseFilterShader()](https://beta.p5js.org/reference/p5/baseFilterShader)
- [baseMaterialShader()](https://beta.p5js.org/reference/p5/baseMaterialShader)
- [baseNormalShader()](https://beta.p5js.org/reference/p5/baseNormalShader)
- [baseStrokeShader()](https://beta.p5js.org/reference/p5/baseStrokeShader)

The point shader, while it has some hooks, is currently still private, as we may end up using the stroke shader to render points in the future.

Reply:
I think `getPointSize` itself is only present in the base point shader, so I think we don't need this one.